rainbow-me/rainbow
# Fix stuck WalletProfileState

Fixes RNBW-4424
Figma link (if any):

## What changed (plus any additional context for devs)
I noticed that in this case the callback closing modal is not called, because we're using a bit different route (creation, where previously we were just caring about deriving). I added the callback invocation in this place as well under gated under additional argument) 


## Screen recordings / screenshots
Before:


https://user-images.githubusercontent.com/25709300/187914905-6539ac1b-258d-43a0-9ef5-ae5e082af335.mov


After:

https://user-images.githubusercontent.com/25709300/187914934-f95bf3c0-bae7-48e6-99d4-155b43780b07.mov



## What to test

Happy scenario:
1. Import a wallet via public key / ens
2. Create a wallet from the wallet selection sheet


Other scenarios:
Check all usages of this modal, namely:
1. Importing via seeds at the beginning 
2. Creating a derived wallet 

Check also if RNBW-4247 is gone
